include/zcbor_print.h

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZCBOR_PRINT_H__
+#define ZCBOR_PRINT_H__
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef ZCBOR_PRINT_FUNC
+#include <stdio.h>
+#define zcbor_do_print(...) printf(__VA_ARGS__)
+#else
+#define zcbor_do_print(...) ZCBOR_PRINT_FUNC(__VA_ARGS__)
+#endif
+
+#ifdef ZCBOR_VERBOSE
+#define zcbor_trace() (zcbor_do_print("bytes left: %zu, byte: 0x%x, elem_count: 0x%zx, err: %d, %s:%d\n",\
```

Comment:
I think it should have a `state` param.

Reply:
Added two new commits where this (and more) is changed. The previous commits are unchanged.